crates/bevy_ecs/src/query/access.rs

```diff
@@ -325,6 +318,84 @@ impl<T: SparseSetIndex> FilteredAccess<T> {
     }
 }
 
+// A struct to express something like `Or<(With<A>, With<B>)>`.
+// Filters like `(With<A>, Or<(With<B>, With<C>)>` are expanded into `Or<((With<A>, With<B>), (With<A>, With<C>))>`.
+#[derive(Clone, Eq, PartialEq)]
+struct ExpandedOrAccess<T, I> {
+    arr: SmallVec<[FixedBitSet; 8]>,
```

Comment:
I'd like to see more docs on exactly how this data structure works. Why do we have multiple fixed bitsets? How does it interact?

Reply:
(the implementation evolved a bit, and so did the comments, resolving it in favour of comments to the up-to-date diff)